### DISCUSSION TITLE
Duplicating a Batch Object

### DISCUSSION THREAD

**Discussion by dloveland** [pyg-team/pytorch_geometric#2268] [Q&A]
Hello, hoping to get some help on a quick question. I am trying to create a duplicate of a Batch object with seperate gradient information. I have seen the problem pop up before in regular PyTorch tensors where calling just .clone() will cause the cloned tensor to share gradients with the original object. Normally this is solved through .detach(), but the Batch object doesn't have this built in. If I want a clean copy, do I need to clone a Batch object and manually call .detach() on all of the tensors within it? Is there a better way to create a copy of Batch object? 

Thanks!

**Comment by rusty1s** [comment]
Great question. Duplicating `data` objects can be achieved via:
```python
copied_data = data.clone()
```
In case you also want to `detach` all tensor attributes after cloning, you can simply run:
```python
copied_data.apply(lambda x: x.detach())
```

**Reply by dloveland** [reply]
Wonderful, thank you for the help!
